### PR TITLE
feat: integrate question filters api to fetch categories

### PR DESCRIPTION
**Pull request by joelwongjy**
- Create query to get question filters from question-service
- Add logic to conditionally display question categories based on user selected difficulties
- Add logic to remove corresponding question categories when user removes difficulties
- Update one incorrect env vars on API gateway

### CODE REVIEW

File: frontend/app/src/features/matching/api/useGetQuestionFilters.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+const getQuestionFiltersSchema = makeSuccessResponseSchema(
+  z.object({
+    Easy: z.array(z.string()),
+    Medium: z.array(z.string()),
+    Hard: z.array(z.string()),
+  }),
+);
+
+const getQuestionFilters = async () => {
+  const { data } = await backendApi.get(API_ENDPOINT.QUESTIONS_GET_FILTERS);
```
Comment (raynerljm) [inline frontend/app/src/features/matching/api/useGetQuestionFilters.ts:19]: the `QUESTIONS_GET_FILTERS` constant is missing a `/` at the start

File: frontend/app/src/features/matching/api/useGetQuestionFilters.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import { makeSuccessResponseSchema } from "@/lib/api";
+import { useQuery } from "react-query";
+import { z } from "zod";
+import { API_ENDPOINT } from "@/constants/api";
+
+const GET_QUESTION_FILTERS = "question-filters";
+
+const getQuestionFiltersSchema = makeSuccessResponseSchema(
+  z.object({
+    Easy: z.array(z.string()),
```
Comment (raynerljm) [inline frontend/app/src/features/matching/api/useGetQuestionFilters.ts]:
```suggestion
    [DIFFICULTY.EASY]: z.array(z.string()),
```

File: frontend/app/src/features/matching/components/SelectPreferencesCard.tsx [hunk 3]
```diff
@@ -44,6 +79,39 @@ export const SelectPreferencesCard = ({ joinCallback }: Props) => {
...
+    }));
+  }, []);
+
+  const topicOptions = useMemo(
+    () => getTopicsForDifficulty(watchDifficulties, data?.data),
+    [watchDifficulties],
+  );
+
+  // Filter user selected topics when difficulty changes
+  useEffect(() => {
```
Comment (raynerljm) [inline frontend/app/src/features/matching/components/SelectPreferencesCard.tsx]: nit: not v good to use a `useEffect` like that. the 'effect' (to update the value) should be set in the change handler instead.

but tbh if it works idrc LOL up to u

**Review by raynerljm — APPROVED** [review]
lgtm! 🎉  except 1 small fix for the API endpoint constant

File: frontend/app/src/features/matching/api/useGetQuestionFilters.ts (hunk 1 above)
Comment (joelwongjy) [reply frontend/app/src/features/matching/api/useGetQuestionFilters.ts:19]: 1c384783e3033a0b73dc3d9dc28efae685db3a86

File: frontend/app/src/features/matching/api/useGetQuestionFilters.ts (hunk 2 above)
Comment (joelwongjy) [reply frontend/app/src/features/matching/api/useGetQuestionFilters.ts]: 1c384783e3033a0b73dc3d9dc28efae685db3a86

File: frontend/app/src/features/matching/components/SelectPreferencesCard.tsx (hunk 3 above)
Comment (joelwongjy) [reply frontend/app/src/features/matching/components/SelectPreferencesCard.tsx]: got rid of useeffect just using a const now 1c384783e3033a0b73dc3d9dc28efae685db3a86

**Pull request merged**